tests/PrefixCacheBootstrapperTest.php

```diff
@@ -0,0 +1,345 @@
+
+    $this->app->singleton(CacheService::class);
+
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($centralValue = 'central-value');
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+
+    tenancy()->initialize($tenant1);
+
+    expect(cache('key'))->toBeNull();
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant1->getTenantKey());
+
+    tenancy()->initialize($tenant2);
+
+    expect(cache('key'))->toBeNull();
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant2->getTenantKey());
+
+    tenancy()->end();
+    expect(cache('key'))->toBe($centralValue);
+});
+
+test('stores not specified in tenantCacheStores do not get prefixed', function() {
+    config(['cache.stores.redis2' => config('cache.stores.redis')]);
+    config(['cache.default' => 'redis2']);
+    // Make 'redis' the only store in $tenantCacheStores so that the current store ('redis2') doesn't get prefixed
+    PrefixCacheTenancyBootstrapper::$tenantCacheStores = ['redis'];
+
+    $this->app->singleton(CacheService::class);
+
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe('central-value');
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+    tenancy()->initialize($tenant1);
+
+    // The cache isn't prefixed, so it isn't separated
+    expect(cache('key'))->toBe('central-value');
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant1->getTenantKey());
+
+    tenancy()->initialize($tenant2);
+
+    expect(cache('key'))->toBe($tenant1->getTenantKey());
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant2->getTenantKey());
+
+    tenancy()->end();
+    expect(cache('key'))->toBe($tenant2->getTenantKey());
+});
+
+test('non default stores get prefixed too', function () {
+    config(['cache.stores.redis2' => config('cache.stores.redis')]);
+    // Make 'redis2' the default cache driver
+    config(['cache.default' => 'redis2']);
+    PrefixCacheTenancyBootstrapper::$tenantCacheStores = ['redis', 'redis2'];
+
+    // The prefix is the same for both drivers in the central context
+    $tenant = Tenant::create();
+    $defaultPrefix = cache()->store()->getPrefix();
+
+    expect(cache()->store('redis')->getPrefix())->toBe($defaultPrefix);
+
+    tenancy()->initialize($tenant);
+    $expectedPrefix = (PrefixCacheTenancyBootstrapper::$prefixGenerator)($tenant);
+
+    // We didn't add a prefix generator for our 'redis' driver, so we expect the prefix to be generated using the 'default' generator
+    expect(cache()->store()->getPrefix())->toBe($expectedPrefix . ':');
+    // Non-default store
+    expect(cache()->store('redis')->getPrefix())->toBe($expectedPrefix . ':');
+
+    tenancy()->end();
+});
+
+test('cache store prefix generation can be customized', function() {
```

Comment:
This test and the next one seem to be using the outdated concept of specific prefix generators for individual cache stores. Now we only have one generator, so probably:
- update the names of the tests
- instead of `expect(cache()->store('redis')->getPrefix())` (direct `store()` call) do what you're doing in `cache base prefix is customizable` where you check both the cache prefix & the cache store prefix.

Also might be worth using multiple cache stores in these tests (including `cache base prefix is customizable`) because I think calling `cache()->getPrefix()` is just an indirect way of getting the prefix from the default store, so the direct call to the store is redundant. Instead it'd make more sense to have multiple stores and to check that the prefix is set on `cache()->getPrefix()` **and each store**. Not 100% sure but I think it'd make more sense that way.

Reply:
> update the names of the tests

The current name of the first test ("cache store prefix generation can be customized") seems fine to me, agree with the next test's name being inaccurate though

> I think calling cache()->getPrefix() is just an indirect way of getting the prefix from the default store, so the direct call to the store is redundant

Right – `cache()->store()->getPrefix()` is the same as `cache()->getPrefix()`

Reply:
I think we can resolve this now, @stancl responded to this on Discord and I updated the code accordingly